### ISSUE TITLE
Allow page files to be organized by domain

### ISSUE THREAD

**Issue by brillout** [brillout/vite-plugin-ssr#125]
For example

```
auth/pages/index.page.js
product/pages/index.page.js
```

**Comment by brillout** [comment]
# Full-fleged example

```shell
# All marketing pages
shop-window/pages/index/index.page.js
shop-window/pages/about/index.page.js
shop-window/pages/jobs/index.page.js
shop-window/_default.page.route.js
# All auth related code 
auth/pages/signup/index.page.js
auth/pages/login/index.page.js
auth/views/UserInfo.js
auth/db/fetchUser.js
# Code for product pages
product/pages/index/index.page.js
product/pages/index/index.page.route.js
product/pages/list/index.page.js
product/pages/list/index.page.route.js
product/db/fetchProduct.js
product/db/fetchProductList.js
# Code common to all pages (the usual `_default` stuff)
_default/_default.page.server.js
_default/_default.page.client.js
_default/views/Header.js
_default/views/Footer.js
_default/images/logo.svg
_default/font/Inter-Var.ttf
```

```js
// shop-window/_default.page.route.js

// New feature: determine the base path of Filesystem Routing
const basePath = '/'
export { basePath }
```
```js
// product/pages/index/index.page.route.js

export default '/product/:productId'
```
```js
// product/pages/list/index.page.route.js

export default '/products'
```

`vite-plugin-ssr` would then route like this:

```
FILESYSTEM                                      URL
shop-window/pages/index/index.page.js           /
shop-window/pages/about/index.page.js           /about
shop-window/pages/jobs/index.page.js            /jobs
auth/pages/signup/index.page.js                 /auth/signup
auth/pages/login/index.page.js                  /auth/login
product/pages/index/index.page.js               /product/:productId
product/pages/list/index.page.js                /products
```

**Comment by brillout** [comment]
If you want this, let me know and I'll implement it.

**Comment by Patrick-Remy** [comment]
This sounds great to me, and would be another argument to switch from Nuxt, as their `modules` are too restricted and not as intuitive as your approach! Thanks for your great work on this plug-in.

**Comment by brillout** [comment]
Done. I will release tomorrow.

**Comment by brillout** [comment]
Tmrw. (Going to be released with a major semver bump and I want to include further breaking changes.)

**Comment by brillout** [comment]
Released in `0.3.1`.
